### DISCUSSION TITLE
Embeddable REPL

### DISCUSSION THREAD

**Discussion by adamgoose** [deepnoodle-ai/risor#101] [Ideas]
I'd like the ability to use Risor to create an embedded repl for my CLI app (which also uses `github.com/spf13/cobra`). Due to a nasty combination of #100 and `internal/cfg` being in an internal package, I'm currently unable to embed the integrated repl.

I'd also like the ability to add additional Modules to the runtime. Since the context of this embedded repl is specific to my application, it'd be handy to expose various Go libraries implemented by my project to the repl as well.

Is any of this currently possible?

**Comment by myzie** [comment]
Hey there @adamgoose,

I’d definitely like Risor to support this. I hadn’t considered wanting to embed the REPL in another project but that sounds handy.

I’ll review internal/cfg and will potentially move it out of internal. This has come up in other projects.

Some of the versioning is complicated by the fact that Risor was migrated from a project called Tamarin which I had in a 1.0.X version and go modules is still getting mixed up on this because of the GitHub project redirect. I think I just need to do a 1.X release of Risor and we’ll be in a better place.

I’ll work on these in the next few days.
